src/Standards/Squiz/Sniffs/Classes/SelfMemberReferenceSniff.php

```diff
@@ -54,7 +54,7 @@ protected function processTokenWithinScope(File $phpcsFile, $stackPtr, $currScop
             }
         }
 
-        if ($conditionToken !== $currScope) {
+        if (isset($conditionToken) === false || $conditionToken !== $currScope) {
```

Comment:
Condition is unnecessary. There will always be a `$conditionToken` set as otherwise the `AbstractScopeSniff` wouldn't have kicked in in the first place.